lib/rspec/core/formatters.rb

```diff
@@ -196,8 +196,8 @@ def built_in_formatter(key)
     end
 
     def notifications_for(formatter_class)
-      formatter_class.ancestors.inject(Set.new) do |notifications, klass|
-        notifications + Loader.formatters.fetch(klass) { Set.new }
+      formatter_class.ancestors.inject(::RSpec::Core::LookupSet.new) do |notifications, klass|
+        notifications + Loader.formatters.fetch(klass) { ::RSpec::Core::LookupSet.new }.to_a
```

Comment:
I'm realizing that the use of `+` here is less efficient than using `merge`, since it allocates an extra object per block invocation.  Thoughts on switching?